### ISSUE TITLE
MC_STATE: don't overload connection and application Reason code spaces

### ISSUE THREAD

**Issue by LPardue** [GrumpyOldTroll/draft-jholland-quic-multicast#84]
The MC_STATE `Reason` field provides a shared code space for connection and application errors. This is quite different from how RFC 9000 does things, whereby CONNECTION_CLOSE indicates connection or application errors via the Frame type. This separation allows all transport concerns to be managed by one set of experts, while delegating application concerns to a different set of experts. That scales well.

I don't see much strong reason to deviate from the design QUIC already uses. The example in https://www.ietf.org/archive/id/draft-jholland-quic-multicast-01.html#section-12.1.1 highlights that an H3 implementation would need to model another error code `0x1000108` somehow, and it would be awkward to figure out where that actually gets registered.

So in summary, I suggest just defining two types of MC_STATE frame
